### PR TITLE
Fix Swinv2ForImageClassification NaN output

### LINKED ISSUE #28675: Swinv2ForImageClassification often outputs NaN at initialization
### System Info

- `transformers` version: 4.36.2
- Platform: Linux-5.4.0-164-generic-x86_64-with-glibc2.31
- Python version: 3.10.11
- Huggingface_hub version: 0.20.2
- Safetensors version: 0.4.1
- Accelerate version: 0.26.1
- Accelerate config:    not found
- PyTorch version (GPU?): 2.0.1 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```py
from transformers import Swinv2Config, Swinv2ForImageClassification
import torch

with torch.inference_mode():
    cfg = Swinv2Config(image_size=56)
    model = Swinv2ForImageClassification(cfg)

    out = model(torch.rand(1, 3, 56, 56))

out.logits
```

### Expected behavior

Should not output NaN; throw an error if the implementation doesn't currently support a certain combination of architectural hyperparameters

### PR DESCRIPTION
**Pull request by miguelm-almeida**
# What does this PR do?

To address the issue of NaN logit outputs for certain combinations of the `image_size`, `patch_size` and `depths` configuration parameters, an assertion was made to ensure that the resulting `window_size` field in the model's Self Attention class is greater than 1, preventing divisions by zero in the normalization of `relative_coords_table`.

Fixes: #28675

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@amyeroberts

### CODE REVIEW

**Comment by NielsRogge** [comment]
Hi,

Thanks for your PR! Could you run `make fixup` so that the CI will be green? Thanks!

**Comment by miguelm-almeida** [comment]
Hi @NielsRogge !

I've been running through this error while running `make fixup`:

image

The files under `src/transformers/models/auto` are all being modified when I run this command and this error keeps going even when I change everything back.

As I am developing on windows, I have made sure that I set `git config core.autocrlf=input`, but can this be related to a LF-CRLF conversion? (It's the only think I can think of, as these files are now marked as modified but I see no changes in the code whatsoever)

Thank you!

**Comment by miguelm-almeida** [comment]
I have also run the `make fix-copies` command as suggested by the CI log:

image

This runs fine but the `make fixup` comes back to the same error after I run it. 
I have also not altered any code regarding `test_modeling_qwen2_moe.py` so I don't know what can be causing this issue.

**Comment by ArthurZucker** [comment]
Rebasing on main should fix this issue

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing this!
